src/array.ts

```diff
@@ -14,6 +16,68 @@ export const group = <T, Key extends string | number | symbol>(
   }, {} as Record<Key, T[]>)
 }
 
+/**
+ * Creates an array of grouped elements, the first of which contains the
+ * first elements of the given arrays, the second of which contains the
+ * second elements of the given arrays, and so on.
+ *
+ * Ex. const zipped = zip(['a', 'b'], [1, 2], [true, false]) // [['a', 1, true], ['b', 2, false]]
+ */
+export function zip<T1, T2, T3, T4, T5>(
+  array1: T1[],
+  array2: T2[],
+  array3: T3[],
+  array4: T4[],
+  array5: T5[]
+): [T1, T2, T3, T4, T5][]
+export function zip<T1, T2, T3, T4>(
+  array1: T1[],
+  array2: T2[],
+  array3: T3[],
+  array4: T4[]
+): [T1, T2, T3, T4][]
+export function zip<T1, T2, T3>(
+  array1: T1[],
+  array2: T2[],
+  array3: T3[]
+): [T1, T2, T3][]
+export function zip<T1, T2>(array1: T1[], array2: T2[]): [T1, T2][]
+export function zip<T>(...arrays: T[][]): T[][] {
+  if (!arrays || !arrays.length) return []
+  return new Array(Math.max(...arrays.map(({ length }) => length)))
+    .fill([])
+    .map((_, idx) => arrays.map(array => array[idx]))
+}
+
+/**
+ * Creates an object mapping the specified keys to their corresponding values
+ *
+ * Ex. const zipped = zipToObject(['a', 'b'], [1, 2]) // { a: 1, b: 2 }
+ */
+export function zipToObject<K extends string | number | symbol, V>(
+  keys: K[],
+  values: V | ((key: K, idx: number) => V) | V[]
+): Record<K, V> {
+  if (!keys || !keys.length) {
+    return {} as Record<K, V>
+  }
+
+  let getValue: (key: K, idx: number) => V
```

Comment:
The bible says `let` and `var` are a sin... not the real bible, the one I wrote. Can we find a way to do this thats non-mutable? Maybe a ternary, maybe an iife, maybe break the reduce down to have if statements.

**1. Ternary**
I'm guessing there were type issues with this one? 
```ts
const getValue = isFunction(values) 
    ? values 
    : isArray(values) 
      ? (k, i) => values[i] 
      : () => values
```

**2. Iife**

```ts
const getValue = (() => {
  if (isFunction(values)) return values
  if (isArray(values)) return (k, i) => values[i]
  return () => vlaues
})()
```

**3. break into the reduce**

obviously this is less optimized since you do checks in every iteration, just trying to give options tho.

```ts
return keys.reduce(
    (acc, key, idx) => {
        if (isFunction(values)) return { ...acc, [key]: values(key, idx) }
        if (isArray(values)) return { ...acc, [key]: values[i] }
        return { ...acc, [key]: values }
    },
    {} as Record<K, V>
  )
```

Reply:
Oooook then I'll just keep the ternary 😅